core/src/test/kotlin/in/specmatic/conversions/OpenApiSpecificationTest.kt

```diff
@@ -3487,7 +3487,86 @@ paths:
 
     }
 
-    @Test
+  @Test
+  fun `support query parameter as free-form dictionary`() {
+    val openAPI =
+      """
+---
+openapi: 3.0.1
+info:
+  title: API
+  version: 1
+paths:
+  /data:
+    get:
+      parameters:
+      - name: userGroups
+        in: query
+        schema:
+          type: object
+          additionalProperties:
+            type: string
+      responses:
+        200:
+          description: API
+""".trimIndent()
+
+    println(openAPI)
+    val feature = OpenApiSpecification.fromYAML(openAPI, "").toFeature()
+
+    val request = HttpRequest("GET", "/data")
```

Comment:
@harikrishnan83 good catch. Both the test and implementation were incomplete, but should be fixed now. Note that I'm not confident in the implementation, but it works until there is a better solution?

I missed that because for my purposes I just needed specmatic to load my openapi.yaml without abruptly exiting, but do not yet need the stubbed endpoint to actually work since my contract testing does not exercise all of the endpoints in stubbed service. It seems like specmatic should be more resilient in this regard, stubbing what it can and only log errors about what it cannot. Is it worth creating an issue for this?

Reply:
@westse sorry we were able to get back to you earlier on this PR. We are looking into it now. Thanks. cc @joelrosario